ztag/schema.py

```diff
@@ -599,50 +602,51 @@ def ztag_zgrab2_transformed(service, results):
     "authentication_mode": zgrab2.postgres.postgres_auth_mode["mode"] | remove_strings,
     "backend_key_data": zgrab2.postgres.postgres_key_data | remove_strings,
-    "tls": ztag_tls_type(doc="If the server allows upgrading the "
+    "tls": probe_tls_type(doc="If the server allows upgrading the "
                              "session to use TLS, this is the log of "
                              "the handshake.")
 }))
 
-
 ztag_schemas = [
-    ("ztag_https", ztag_tls),
-    ("ztag_heartbleed", ztag_heartbleed),
-    ("ztag_smtp_starttls", ztag_smtp_starttls),
-    ("ztag_imap_starttls", ztag_mail_starttls),
-    ("ztag_pop3_starttls", ztag_mail_starttls),
-    ("ztag_imap_tls", ztag_mail_tls),
-    ("ztag_pop3_tls", ztag_mail_tls),
-    ("ztag_http", ztag_http),
-    ("ztag_ftp", ztag_ftp),
-    ("ztag_dh", ztag_dh),
-    ("ztag_dh_export", ztag_dh_export),
-    ("ztag_rsa_export", ztag_rsa_export),
-    ("ztag_ecdh", ztag_ecdh),
-    ("ztag_sslv2", ztag_sslv2),
-    ("ztag_sslv3", ztag_tls_support),
-    ("ztag_tls1", ztag_tls_support),
-    ("ztag_tls2", ztag_tls_support),
-    ("ztag_tls3", ztag_tls_support),
-    ("ztag_modbus", ztag_modbus),
-    ("ztag_extended_random", ztag_extended_random),
-    ("ztag_ssh_v2", ztag_ssh_v2),
-    ("ztag_dns_lookup", ztag_dns_lookup),
-    ("ztag_bacnet", ztag_bacnet),
-    ("ztag_fox", ztag_fox),
-    ("ztag_dnp3", ztag_dnp3),
-    ("ztag_s7", ztag_s7),
-    ("ztag_smb", ztag_smb),
-    ("ztag_upnp_discovery", ztag_upnp_discovery),
+    ("ztag_https", probe_tls_type()),
+    ("ztag_heartbleed", probe_tls_heartbleed_type()),
+    ("ztag_smtp_starttls", probe_smtp_starttls_banner_type()),
+    ("ztag_imap_starttls", probe_mail_starttls_banner_type()),
+    ("ztag_pop3_starttls", probe_mail_starttls_banner_type()),
+    ("ztag_imap_tls", probe_mail_tls_banner_type()),
+    ("ztag_pop3_tls", probe_mail_tls_banner_type()),
+    ("ztag_http", probe_http_request_type()),
+    ("ztag_ftp", probe_ftp_banner_type()),
+    ("ztag_dh", probe_tls_dh_type()),
+    ("ztag_dh_export", probe_tls_dh_export_type()),
+    ("ztag_rsa_export", probe_tls_rsa_export_type()),
+    ("ztag_ecdh", probe_tls_ecdh_type()),
+    ("ztag_sslv2", probe_tls_sslv2_type()),
+    ("ztag_sslv3", probe_tls_support_type()),
+    ("ztag_tls1", probe_tls_support_type()),
+    ("ztag_tls2", probe_tls_support_type()),
+    ("ztag_tls3", probe_tls_support_type()),
+    ("ztag_modbus", probe_modbus_device_id_type()),
+    ("ztag_extended_random", probe_tls_extended_random_type()),
+    ("ztag_ssh_v2", probe_ssh_v2_type()),
+    ("ztag_dns_lookup", probe_dns_lookup_type()),
+    ("ztag_bacnet", probe_bacnet_device_id_type()),
+    ("ztag_fox", probe_fox_device_id_type()),
+    ("ztag_dnp3", probe_dnp3_status_type()),
+    ("ztag_s7", probe_s7_szl_type()),
+    ("ztag_smb", probe_smb_banner_type()),
+    ("ztag_upnp_discovery", probe_upnp_discovery_type()),
     ("ztag_oracle", ztag_oracle),
```

Comment:
Was there any reason these were left alone?

Reply:
Those are produced by a transform from zgrab schemas, which I haven't figured out how to handle yet.  Do we want to abandon zgrab schemas and rip them out?  Duplicate the code and edit from there?  Edit in zgrab and continue to have the complexity of the transform?  Would love to get everyone's thoughts.

Reply:
As I understand it, once we get to the point where zgrab2's output is defined via the protobuf generated by the schema, the zgrab schemas will be moot.

For now though, referencing the zgrab schemas (as well as the zcrypto schemas) seems like the simplest way to avoid duplicating code and the associated maintenance nightmares (further, the reason ztag references zgrab rather than vice-versa is because the schemas were reverse engineered from the zgrab JSON output -- whereas once we have the zgrab JSON output defined by the schemas, both the duplication problem and dependency reversal disappear).